src/writer.rs

```diff
@@ -8,6 +8,33 @@ use tractogram::{Point, RefTractogramItem, Tractogram, TractogramItem};
 use {Affine, Affine4, CHeader, Header, Translation};
 use affine::get_affine_and_translation;
 
+macro_rules! write_streamline {
+    ($writer:ident, $streamline:expr, $scalars:expr, $properties:expr) => {
+        if $writer.nb_scalars == 0 {
+            $streamline.write($writer);
+        } else {
+            $writer.writer.write_i32::<LittleEndian>($streamline.len() as i32).unwrap();
+            $writer.real_n_count += 1;
+
+            let scalars = $scalars.chunks($writer.nb_scalars);
+            for (p, scalars) in $streamline.into_iter().zip(scalars) {
+                $writer.write_point(&p);
+                $writer.write_sp(scalars);
+            }
+        }
+
+        $writer.write_sp($properties);
+    };
+    // Fast method, without scalars and properties
+    ($writer:ident, $streamline:expr, $nb_points:expr) => {
+        $writer.writer.write_i32::<LittleEndian>($nb_points as i32).unwrap();
```

Comment:
Why are you not using `$streamline.len()` here? Why do we need `nb_points`?

Reply:
Because this macro can also be used by `write_from_iter`, which receives an iterator. We don't know in advance how many elements the iterator will produce, so the caller gives the lenght.

Reply:
Would it be possible to have a version that doesn't take `nb_points` when we're in the non-iterator case?. `write!(streamline)`? Thoughts?

Reply:
It is possible, yes, but I don't see the point because this new version would add a 7 lines code copy and it will only change one line from

    write_streamline!(writer, self, self.len());
    // to
    write_streamline!(writer, self);

IMO, it's not worth it.